classes/submission/Collector.php

```diff
@@ -288,22 +303,44 @@ public function getQueryBuilder(): Builder
             case self::ORDERBY_TITLE:
                 $locale = Locale::getLocale();
                 $q->leftJoin('publications as publication_tlp', 's.current_publication_id', '=', 'publication_tlp.publication_id')
-                    ->leftJoin('publication_settings as publication_tlps', function (JoinClause $join) use ($locale) {
+                    ->leftJoin('publication_settings as publication_tlps', fn (JoinClause $join) =>
                         $join->on('publication_tlp.publication_id', '=', 'publication_tlps.publication_id')
                             ->where('publication_tlps.setting_name', '=', 'title')
                             ->where('publication_tlps.setting_value', '!=', '')
-                            ->where('publication_tlps.locale', '=', $locale);
-                    });
+                            ->where('publication_tlps.locale', '=', $locale)
+                    );
                 $q->leftJoin('publications as publication_tlpl', 's.current_publication_id', '=', 'publication_tlpl.publication_id')
-                    ->leftJoin('publication_settings as publication_tlpsl', function (JoinClause $join) {
+                    ->leftJoin('publication_settings as publication_tlpsl', fn (JoinClause $join) =>
                         $join->on('publication_tlp.publication_id', '=', 'publication_tlpsl.publication_id')
                             ->on('publication_tlpsl.locale', '=', 's.locale')
-                            ->where('publication_tlpsl.setting_name', '=', 'title');
-                    });
+                            ->where('publication_tlpsl.setting_name', '=', 'title')
+                    );
                 $coalesceTitles = 'COALESCE(publication_tlps.setting_value, publication_tlpsl.setting_value)';
                 $q->addSelect([DB::raw($coalesceTitles)]);
                 $q->orderBy(DB::raw($coalesceTitles), $this->orderDirection);
                 break;
+            case self::ORDERBY_SEARCH_RANKING:
+                if (!$keywords->count()) {
+                    $q->orderBy('s.date_submitted', $this->orderDirection);
+                    break;
+                }
+                // Retrieves the number of matches for all keywords
+                $orderByMatchCount = DB::table('submission_search_objects', 'sso')
```

Comment:
Comment for here and below, a single query using `keyword IN (x, y, z)` becomes very slow if one of the values doesn't exist (I tested locally and on a production database with a large dataset) o_O